examples/jsm/loaders/PLYLoader.js

```diff
@@ -343,45 +347,53 @@ class PLYLoader extends Loader {
 
 			let result;
 
-			const patternBody = /end_header\s([\s\S]*)$/;
+			const patternBody = /end_header\s+([\s\S]*)$/;
 			let body = '';
 			if ( ( result = patternBody.exec( data ) ) !== null ) {
 
 				body = result[ 1 ];
 
 			}
 
-			const lines = body.split( /\r\n|\r|\n/ );
-			let currentElement = 0;
-			let currentElementCount = 0;
-			let elementDesc = header.elements[ currentElement ];
-			let attributeMap = mapElementAttributes( elementDesc.properties );
-
-			for ( let i = 0; i < lines.length; i ++ ) {
+			class TokenStream {
```

Comment:
Do you mind moving this class definition outside of `PLYLoader` into the module scope? Maybe right above the export statement...